### PR TITLE
HOTT-1079 Fixes on switch banner

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-1079

### What?
- This fixes the double switch banner issue and hides the banner on the Meursing code finder page (and other pages).
- Also, this makes a small refactor on the shared/search partial

### Why?
Because we hate BUGS!!! :rage1:

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -73,8 +73,8 @@ def set_last_updated
     # rubocop:enable Naming/MemoizedInstanceVariableName
   end
 
-  def set_enable_service_switch_banner_in_action
-    @enable_service_switch_banner_in_action = true
+  def set_switch_banner
```
Comment (jebw) [inline app/controllers/application_controller.rb]: I know I'm going to seem really picky here - but `switch_banner` doesn't make any reference to what its switching - a potential point of confusion for future developers.

Can this be renamed to something like `service_switch_banner` which is a bit more specific but still short?

**Review by jebw — COMMENTED** [review]
Only comment is around naming?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -76,11 +76,13 @@
       <%= render 'shared/phase_banner' if ENV['NOTICE_BANNER'] %>
       <%= yield :before_main_content %>
 
+      <%= render 'shared/top_breadcrumbs' %>
+      <%= render 'shared/switch_banner' %>
```
Comment (alexdesi) [inline app/views/layouts/application.html.erb:80]: Those partials have been extracted from the `shared/search` partial.
(why did the search contain the breadcrumb and switch? 🤷 )

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (alexdesi) [reply app/controllers/application_controller.rb]: 👍

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -73,8 +73,16 @@ def set_last_updated
     # rubocop:enable Naming/MemoizedInstanceVariableName
   end
 
-  def set_enable_service_switch_banner_in_action
-    @enable_service_switch_banner_in_action = true
+  def set_switch_service_banner
```
Comment (willfish) [inline app/controllers/application_controller.rb]:
```suggestion
  def enable_switch_service_banner
```

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@ class ApplicationController < ActionController::Base
   include CookiesHelper
 
   before_action :set_cache
-  before_action :set_enable_service_switch_banner_in_action
+  before_action :set_switch_service_banner
```
Comment (willfish) [inline app/controllers/application_controller.rb]:
```suggestion
  before_action :enable_switch_service_banner
```

File: app/controllers/pages_controller.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 class PagesController < ApplicationController
   before_action do
     @tariff_last_updated = nil
+
```
Comment (willfish) [inline app/controllers/pages_controller.rb]: Do you want this extra line :)?

File: app/controllers/pages_controller.rb [hunk 6]
```diff
@@ -15,8 +15,8 @@ def opensearch
   end
 
   def tools
+    set_switch_service_banner
```
Comment (willfish) [inline app/controllers/pages_controller.rb]:
```suggestion
    enable_switch_service_banner
```

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (jebw) [reply app/controllers/application_controller.rb]: I'll add my +1 to will's suggestion

File: app/controllers/pages_controller.rb (hunk 6 above)
Comment (willfish) [reply app/controllers/pages_controller.rb]: Actually isn't this the default so I don't think you need this?

File: app/controllers/search_controller.rb [hunk 7]
```diff
@@ -46,8 +46,6 @@ def quota_search
 
     form = QuotaSearchForm.new(params)
     @result = QuotaSearchPresenter.new(form)
-    @enable_service_switch_banner_in_action = false
```
Comment (willfish) [inline app/controllers/search_controller.rb:49]: Shouldn't you replace this with a call to `disable_switch_service_banner`

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (willfish) [reply app/views/layouts/application.html.erb:80]: So much better

File: app/controllers/search_controller.rb (hunk 7 above)
Comment (alexdesi) [reply app/controllers/search_controller.rb:49]: Good catch! 
Actually here the banner was enabled here, so I removed that line.

(it has been enabled in various places too)

**Review by willfish — APPROVED** [review]

**Pull request merged**